fixed-point-test.rkt

```diff
@@ -0,0 +1,12 @@
+#lang racket
+
+(require rackunit
+         "fixed-point.rkt")
+
+(define (inc x) (+ x 1))
+(define (dec x) (- x 1))
+(define (identity x) x)
+(define epsilon 0.000001)
+
+(check-equal? (fixed-point identity inc 1 epsilon 10) 11 "NotConverging")
+(check-equal? (fixed-point identity dec 1 epsilon 100) 0 "Converging")
```

Comment:
hmm I don't understand this tests. The identity function does have fixed points (it's the definition), but both calls for _fixed-point_ must not converge in this cases because the _step_ function is always greater than _epsilon_. 
What I'm missing?

Reply:
The second call is "converging" just because (dec 1) equals zero, then lower than epsilon.. but fails for every other value.

Does not your _fixed-point_ implementation depend upon the _next_ function step at a fraction of x lower than epsilon? If so, why this tests?

Reply:
Now that you asked I think I implemented the wrong idea. My fixed point actually searches for 0 (that is why incrementing will never converge), but this is wrong. By definition a fixed point is basically f(x) - f(x+1) < epsilon. Just me being me...stupid =D. Thanks ;-)

Reply:
By definition a fixed point is y = f(x) = x, right? The identity function, by definition, is a function that every point is a fixed point.

Reply:
Again I understood stuff wrong =D

https://en.wikipedia.org/wiki/Fixed_point_(mathematics)

You are right...I may get it right eventually =P